### PR TITLE
wrote merkle root of tree built from execution results to global state

### PR DESCRIPTION
**Pull request by SamuelSchlesinger**
Issue: https://github.com/casper-network/casper-node/issues/3242

### CODE REVIEW

File: types/src/key.rs [hunk 1]
```diff
@@ -816,6 +865,9 @@ impl Distribution<Key> for Standard {
             10 => Key::SystemContractRegistry,
             11 => Key::Unbond(rng.gen()),
             12 => Key::ChainspecRegistry,
+            13 => Key::ExecutionResultsRootHash {
+                block_height: rng.gen(),
+            },
```
Comment (goral09) [inline types/src/key.rs:889]: I'm susprirsed that there's no way to derive `Distribution` for an enum.

File: node/src/components/contract_runtime/operations.rs [hunk 2]
```diff
@@ -97,6 +100,27 @@ pub fn execute_finalized_block(
         state_root_hash = state_hash;
     }
 
+    let execution_results_root_hash = compute_execution_results_root_hash(
+        &mut execution_results.iter().map(|(_, _, result)| result),
+    )?;
+    let block_height = finalized_block.height();
```
Comment (Fraser999) [inline node/src/components/contract_runtime/operations.rs:106]: Looks like we can now remove the same code from line 173 below.

**Review by Fraser999 — COMMENTED** [review]
Other than the Key variant rename discussed offline, and the nitpick, this LGTM.

File: execution_engine_testing/tests/src/test/explorer/faucet.rs [hunk 3]
```diff
@@ -814,7 +814,7 @@ fn faucet_costs() {
     // This test will fail if execution costs vary.  The expected costs should not be updated
     // without understanding why the cost has changed.  If the costs do change, it should be
     // reflected in the "Costs by Entry Point" section of the faucet crate's README.md.
```
Comment (Fraser999) [inline execution_engine_testing/tests/src/test/explorer/faucet.rs:816]: We need to update [the table mentioned in this comment](https://github.com/casper-network/casper-node/tree/dev/smart_contracts/contracts/explorer/faucet#costs-by-entry-point) too.

File: types/src/key.rs [hunk 4]
```diff
@@ -45,6 +46,7 @@ const DICTIONARY_PREFIX: &str = "dictionary-";
 const UNBOND_PREFIX: &str = "unbond-";
 const SYSTEM_CONTRACT_REGISTRY_PREFIX: &str = "system-contract-registry-";
 const CHAINSPEC_REGISTRY_PREFIX: &str = "chainspec-registry-";
+const BLOCK_EFFECTS_ROOT_HASH_PREFIX: &str = "execution-results-root-hash-";
```
Comment (Fraser999) [inline types/src/key.rs]:
```suggestion
const BLOCK_EFFECTS_ROOT_HASH_PREFIX: &str = "block-effects-root-hash-";
```

**Review by Fraser999 — APPROVED** [review]
Couple of final nits, but optimistic approval assuming these will be addressed.

**Comment by Fraser999** [comment]
bors merge

**Pull request merged**